### PR TITLE
rlImGui: added different mouse cursor support

### PR DESCRIPTION
**Pull request by Julianiolo**
#### I added mouse Cursor support
its inspired/based by/on the win32 backend from Dear ImGui

it implements all cursors (I think)

#### Changes
I changed the original part that hid and unhid the cursor and made it only change it if the cursor is changed (or ImGui draws the cursor itself)

### CODE REVIEW

File: rlImGui/rlImGui.cpp [hunk 1]
```diff
@@ -48,6 +50,31 @@ static void rlImGuiSetClipText(void*, const char* text)
     SetClipboardText(text);
 }
 
+static bool RLImGuiSetMouseCursor(ImGuiMouseCursor imguiMouseCursor)
+{
+	ImGuiIO& io = ImGui::GetIO();
+
+	if (io.ConfigFlags & ImGuiConfigFlags_NoMouseCursorChange) {
```
Comment (JeffM2501) [inline rlImGui/rlImGui.cpp]: newline before the { please.

File: rlImGui/rlImGui.cpp [hunk 2]
```diff
@@ -48,6 +50,31 @@ static void rlImGuiSetClipText(void*, const char* text)
...
+		return false;
+	}
+	
+	int cursor;
+	switch (imguiMouseCursor) {
```
Comment (JeffM2501) [inline rlImGui/rlImGui.cpp]: newline before the { please

File: rlImGui/rlImGui.cpp [hunk 3]
```diff
@@ -48,6 +50,31 @@ static void rlImGuiSetClipText(void*, const char* text)
...
+		case ImGuiMouseCursor_Arrow:      cursor = MOUSE_CURSOR_ARROW;         break;
+		case ImGuiMouseCursor_TextInput:  cursor = MOUSE_CURSOR_IBEAM;         break;
+		case ImGuiMouseCursor_Hand:       cursor = MOUSE_CURSOR_POINTING_HAND; break;
+		case ImGuiMouseCursor_ResizeAll:  cursor = MOUSE_CURSOR_RESIZE_ALL;    break;
+		case ImGuiMouseCursor_ResizeEW:   cursor = MOUSE_CURSOR_RESIZE_EW;     break;
+		case ImGuiMouseCursor_ResizeNESW: cursor = MOUSE_CURSOR_RESIZE_NESW;   break;
+		case ImGuiMouseCursor_ResizeNS:   cursor = MOUSE_CURSOR_RESIZE_NS;     break;
+		case ImGuiMouseCursor_ResizeNWSE: cursor = MOUSE_CURSOR_RESIZE_NWSE;   break;
+		case ImGuiMouseCursor_NotAllowed: cursor = MOUSE_CURSOR_NOT_ALLOWED;   break;
+		default:                          cursor = MOUSE_CURSOR_DEFAULT;       break;
```
Comment (JeffM2501) [inline rlImGui/rlImGui.cpp]: case, assignment and break on separate lines, or store the enums in a map.

File: rlImGui/rlImGui.cpp [hunk 4]
```diff
@@ -84,10 +111,15 @@ static void rlImGuiNewFrame()
     if ((io.ConfigFlags & ImGuiConfigFlags_NoMouseCursorChange) == 0)
     {
         ImGuiMouseCursor imgui_cursor = ImGui::GetMouseCursor();
-        if (io.MouseDrawCursor || imgui_cursor == ImGuiMouseCursor_None)
-            HideCursor();
-        else
-            ShowCursor();
+		if (imgui_cursor != lastMouseCursor || io.MouseDrawCursor) {
```
Comment (JeffM2501) [inline rlImGui/rlImGui.cpp]: newline before the { please

File: rlImGui/rlImGui.cpp [hunk 5]
```diff
@@ -84,10 +111,15 @@ static void rlImGuiNewFrame()
...
+			lastMouseCursor = imgui_cursor;
+			if (io.MouseDrawCursor || imgui_cursor == ImGuiMouseCursor_None)
+				HideCursor();
```
Comment (JeffM2501) [inline rlImGui/rlImGui.cpp]: if the else has brackets then the if should have brackets too (so they both match, even if the if is one line)

File: rlImGui/rlImGui.cpp [hunk 6]
```diff
@@ -84,10 +111,15 @@ static void rlImGuiNewFrame()
...
+			else {
```
Comment (JeffM2501) [inline rlImGui/rlImGui.cpp]: newline before the { please

File: rlImGui/rlImGui.cpp (hunk 4 above)
Comment (JeffM2501) [reply rlImGui/rlImGui.cpp]: Also indent spacing is wrong.

File: rlImGui/rlImGui.cpp [hunk 7]
```diff
@@ -302,6 +334,8 @@ void SetupRLImGui(bool dark)
     ImGuiIO& io = ImGui::GetIO();
     io.BackendPlatformName = "imgui_impl_raylib";
 
+	io.BackendFlags |= ImGuiBackendFlags_HasMouseCursors;
```
Comment (JeffM2501) [inline rlImGui/rlImGui.cpp]: extra space.

**Review by JeffM2501 — CHANGES_REQUESTED** [review]
Pretty good, just needs some style cleanups to match the existing code.

File: rlImGui/rlImGui.cpp (hunk 1 above)
Comment (Julianiolo) [reply rlImGui/rlImGui.cpp]: Already did that in 2nd commit

File: rlImGui/rlImGui.cpp (hunk 2 above)
Comment (Julianiolo) [reply rlImGui/rlImGui.cpp]: Already did that in 2nd commit

File: rlImGui/rlImGui.cpp (hunk 7 above)
Comment (Julianiolo) [reply rlImGui/rlImGui.cpp]: what do you mean exactly?

Comment (Julianiolo) [reply rlImGui/rlImGui.cpp]: if you are talking about the indenting thats fixed now

**Comment by JeffM2501** [comment]
Thanks for the fix :)

**Pull request merged**